plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -990,6 +990,41 @@ var _ = Describe("validator", func() {
 				Expect(shoot.Spec.Kubernetes.Version).To(Equal(highestPatchVersion.Version))
 			})
 
+			It("should default a major.minor kubernetes version only to non-preview versions", func() {
+				shoot.Spec.Kubernetes.Version = "1.6"
+				preview := core.ClassificationPreview
+				previewVersion := core.ExpirableVersion{Version: "1.6.6", Classification: &preview}
+				highestNonPreviewPatchVersion := core.ExpirableVersion{Version: "1.6.5"}
+				cloudProfile.Spec.Kubernetes.Versions = append(cloudProfile.Spec.Kubernetes.Versions, previewVersion, highestNonPreviewPatchVersion, core.ExpirableVersion{Version: "1.7.1"}, core.ExpirableVersion{Version: "1.7.2"})
+
+				coreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&project)
```

Comment:
Can you please wrap all calls to the informer store with `Expect(...).To(Succeed())` to avoid incorrect tests?
```suggestion
				Expect(coreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&project)).To(Succeed())
```